### ISSUE TITLE
Unable to compile sourcecode

### ISSUE THREAD

**Issue by klaszlo** [Thinkmill/react-conf-app#51]
Hi!

I would like to take your application as a starting point developing 
my own mobile app (kind of hello world on steroids:), but I'm unable to compile it.

I tried with both android studio (2.3), and with react native script:
`./node_modules/.bin/react-native run-android` after installing with `npm -i`

It is basically a full circle, it errors out with SDK version too low, it requires 23.0.3,
after it requires 25.0.0. Then the final error is(react-native command line):
 
    > Could not resolve all dependencies for configuration ':app:_debugApk'.
    > A problem occurred configuring project ':react-native-smart-splash-screen'.
      > The SDK Build Tools revision (23.0.3) is too low for project ':react-native-smart-splash-screen'. Minimum required is 25.0.0

In Android Studio it is installed 
(`File -> settings->Appearance & behaviour -> System settings -> Android SDK->SDK Tools`), 
- [x] 23.0.3
- [x] 25.0.0
- [x] 25.0.2

In Android Studio the error is:

    Error:Failed to find Build Tools revision 23.0.3
    <a href="install.build.tools">Install Build Tools 23.0.3 and sync project</a>

If I remove 25.0.0, then the error is:

     Gradle sync failed: Failed to find Build Tools revision 25.0.0
				Consult IDE log for more details (Help | Show Log)

In the last week I installed and reinstalled android studio all the sdk versions, I used up (on my mobile net:), more then 36GiB of data, still no luck. One try takes more then an hour, I'm on a bit slow connection (4mbit down/2mbit up).

I'm on ubuntu 16.10 by the way.

Could you please tell me if the project is buildable, and if so what version of android studio you used?

Sorry for the offtopic.

**Comment by marukami** [comment]
I just built it. So, yes. 

You need to install multiple version of the Build Tools. 23.0.3, 25.0.0, and 25.0.2 sounds about right for most projects. Tho, you may need more.

Make sure the Android SDK home is in your PATH. Here is a `sh` I use when setting up Mac environments 
```bash
export ANDROID_HOME=$HOME/Library/Android/SDK
export PATH=$ANDROID_HOME/platform-tools:$PATH
export PATH="$PATH:$ANDROID_HOME/platform-tools"
```

**NOTE** Since I'm on Mac, you may need to replace `$HOME/Library` with what every you have in `File -> settings->Appearance & behaviour -> System settings -> Android SDK->SDK Tools`

If all else fails, there are some docker images with everything installed like https://github.com/appunite/docker/tree/master/android-java7-8-r25-0-1 **WARNING** The android-java7-8-r25 docker image is around 10GB. 

https://github.com/heikomaass/docker-android/blob/master/android-sdk/Dockerfile is a bit lighter. ~2GB if I remember correctly. You may need to extend the image and add more build tools.

**Comment by blargity** [comment]
@klaszlo, I'm able to build it as well. Here's some screenshots of the SDK manager in Android studio for me: 
![screen shot 2017-04-04 at 9 39 21 am](https://cloud.githubusercontent.com/assets/1531750/24636042/b350c204-191a-11e7-9946-03738f0de56b.png)
![screen shot 2017-04-04 at 9 39 33 am](https://cloud.githubusercontent.com/assets/1531750/24636045/bb994a94-191a-11e7-8014-212430c6671b.png)

I'll go ahead and close this issue for now, but please let me know if you're still having trouble building, as we want to help!

**Comment by klaszlo** [comment]
Thank you for the help @blargity and @marukami.
And sorry for the late reply. 

I installed and reinstalled numerous time (both android studio, react-conf-app, also removing `~/.gradle` directory).

After I modified gradle version (`android/build.gradle`):

    classpath 'com.android.tools.build:gradle:2.2.3'
    
Correcting a path issue (`android/settings.gradle`):

    project(':react-native-maps').projectDir = new File(rootProject.projectDir, '../node_modules/react-native-maps/lib/android')

I'm finally able to launch the application:
![Uploading react-conf-app_error.png…]()


If I minimize and maximize the application in the emulator, I get the splash screen (ReactConf2017) and not this error message anymore.

Still some hunting is required from my side to get rid of this error (DeviceInfo native module is not installed correctly).

mini update: 
After restarting, the above error magically disappear, and the following appear (the same red background):
> Could not get BatchedBridge, make sure your bundle is packaged correctly.
